services/api-server/src/simcore_service_api_server/services/_service_exception_handling.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from contextlib import contextmanager
+from functools import wraps
+from typing import Mapping
+
+import httpx
+from fastapi import HTTPException, status
+from pydantic import ValidationError
+from servicelib.error_codes import create_error_code
+
+_logger = logging.getLogger(__name__)
+
+
+def service_exception_mapper(
+    service_name: str, http_status_map: Mapping[int, tuple[int, str | None]]
+):
+    def decorator(func):
+        @wraps(func)
+        async def wrapper(*args, **kwargs):
+            with backend_service_exception_handler(service_name, http_status_map):
+                return await func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+@contextmanager
+def backend_service_exception_handler(
+    service_name: str, http_status_map: Mapping[int, tuple[int, str | None]]
+):
+    try:
+        yield
+    except ValidationError as exc:
+        error_code = create_error_code(exc)
+        _logger.exception(
+            "Invalid data exchanged with %s service [%s] ",
+            service_name,
+            error_code,
+            extra={"error_code": error_code},
+        )
+        raise HTTPException(
+            status_code=status.HTTP_502_BAD_GATEWAY,
+            detail=f"{service_name} service returned invalid response. {error_code}",
+        ) from exc
+
+    except AssertionError as exc:
```

Comment:
Do we use AssertionError? AFAIK we do not. NOTE that all the `asssert` in the code are removed in production. WE use `PYTHONOPTIMIZE=1`. SEE  https://github.com/itisfoundation/osparc-simcore/blob/a284681421fbee72fa639ec9f76222eda4f3411e/services/api-server/Dockerfile#L128


[Python 3 Command Line and Environment Documentation](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONOPTIMIZE)